### ISSUE TITLE
The FCS_COP.1/KeyEnc Cryptographic operation (Key Encryption) SFR in HCD cPP v1.0 is inconsistent with TPM 2.0 Architecture specification section "26.6 Sensitive Area Encryption"

### ISSUE THREAD

**Issue by ansukert** [HCD-iTC/HCD-IT#1]
As noted by Alan, Lexmark stumbled into an issue with the HCD cPP v1.0 for the FCS_COP.1/KeyEnc Cryptographic operation (Key Encryption) SFR on page 62:

The SFR states that:
The TSF shall perform key encryption and decryption in
accordance with a specified cryptographic algorithm [selection:
Case: AES algorithm
• AES used in [[selection: CBC, GCM] mode] and cryptographic key sizes [selection: 128 bits, 192
bits, 256 bits] that meet the following: AES as specified in ISO/IEC 18033-3, [selection: CBC as
specified in ISO/IEC 10116, GCM as specified in ISO/IEC 19772],

However, for this particular Lexmark product the original product design utilized a TPM not to store an encryption key, but to encrypt it utilizing the TPM2_Seal() API and then storing the encrypted key in flash outside the TPM.  While documenting our compliance with FPT_KYP_EXT.1.1 we discovered that while the TPM was compliant with the TPM 2.0 Architecture specification section "26.6 Sensitive Area Encryption", it was violation of the HCD cPP FCS_COP.1.1/KeyEnc requirements, since it did not use it use CBC or GCM mode.

From TPM 2.0 Architecture specification Section 26.6 (Page 172) https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part1_Architecture_pub.pdf -"All symmetric encryption of the sensitive area uses Cipher Feedback (CFB) mode." NOTE: CFB is the only AES mode allowed by the TPM 2.0 specification for this activity.

PROPOSAL:
  Change the text in A.2.2 FCS_COP.1.1/KeyEnc from:
    ...
    Case: AES algorithm:
    • AES used in [[selection: CBC, GCM] mode] and cryptographic key sizes [selection: 128 bits, 192
    bits, 256 bits] that meet the following: AES as specified in ISO/IEC 18033-3, [selection: CBC as
    specified in ISO/IEC 10116, GCM as specified in ISO/IEC 19772],
    ...
  to:
    ...
    Case: AES algorithm:
    • AES used in [[selection: CBC, GCM, CFB] mode] and cryptographic key sizes [selection: 128 bits, 192
    bits, 256 bits] that meet the following: AES as specified in ISO/IEC 18033-3, [selection: CBC as
    specified in ISO/IEC 10116, GCM as specified in ISO/IEC 19772,  CFB as specified in ISO/IEC 10116],
    ...

In layman's terms, add CFB as an allowed mode to encrypt a key when using the AES algorithm.

**Comment by Ryuichiro-Ohya-FX** [comment]
I can accept Graydon-san’s proposal to add CFB in FCS_COP.1/KeyEnc. But, you all might be aware of it, the case which Graydon-san described in the issue might be the one which was exempted to verify the protection method by FCS_KYC_EXT.1.
In brief, we intended to make the key in TPM out of the key chain in FPT_KYP_EXT.1, and to exempt to verify the protection method if the key in TPM is used for key encryption.
Of course, if CFB is used for key encryption with the key which the application can access, then CFB should be added to the selection.
I’d like to know your intention in the discussion we will make.
[Exemption of protection method verification in FCS_KYC_EXT.pptx](https://github.com/HCD-iTC/HCD-IT/files/11269786/Exemption.of.protection.method.verification.in.FCS_KYC_EXT.pptx)

**Comment by jmcdaniels** [comment]
AES CFB should be an acceptable addition if it is needed.

**Comment by tbenkartidtec** [comment]
Kwangwoo asked how the proposed addition of AES-CFB mode as an allowed selection option in FCS_COP.1.1/KeyEnc in the HCD cPP would correspond to changes in [draft version 0.1 of the Specification of Functional Requirements for Cryptography](https://ccusersforum.onlyoffice.co/Products/Projects/TMDocs.aspx?prjID=599711#3388358) [Crypto] from the Crypto Working Group.

[Crypto] does not include an SFR specific to key encryption (e.g., FCS_COP.1/KeyEnc in the HCD cPP), so there is no direct direct correspondence.  [Crypto] includes iterations of FCS_COP.1 for Key Encapsulation (/KeyEncap) and Key Wrapping (/KeyWrap), but neither of those SFRs involve usage of AES in a mode close to CFB.

However, [Crypto] includes FCS_CKM_EXT.3 for Cryptographic Key Access, which does directly reference key encryption:

FCS_CKM_EXT.3.1 The TSF shall use specified cryptographic key access methods [selection: key encapsulation, key wrapping, **key encryption**] that meets the following: [selection: list of standards] to access keys when performing [selection: cryptographic key usage in cryptographic operations, cryptographic key archival, cryptographic key backup, cryptographic key escrow, cryptographic key recovery, modifications to attributes of cryptographic keys, cryptographic key destruction].

In the Application Note for this SFR, specifically in the section regarding Key Access Operations, typical actions for the life cycle of a key are identified.  The following two actions potentially correspond to our proposed change to FCS_COP.1.1/KeyEnc (with relevant text bolded):

**Cryptographic key usage in cryptographic operations** – Depending on the application, a cryptographic key could be root key which by virtue of being a root key, cannot be encrypted with another key.  It is often stored in plain sight, sometimes with hardware protections, sometimes obfuscated in clever ways, such as in physically unclonable functions.  **Other than root keys, it is expected that other cryptographic keys should be protected with encryption, often using other keys, which leads to a chain or hierarchy of cryptographic keys that encrypt other keys often anchored by a root key.**   Additionally, it is expected that an entity authenticate itself self as an authorized user of the key.  This latter requirement is in addition to the recommended requirements of the Common Criteria Part 2 for FCS_CKM.3.  

**Cryptographic key archival, backup, and escrow** – TOEs often perform cryptographic key archival to manage limited memory resources inside its own security boundaries.  **Often the TOE encrypts the cryptographic keys prior to saving them to storage that is close by on the same device as the TOE and that can be accessed quickly.** TOEs may perform cryptographic key backups into storage that is meant for longer term keeping.  Backup storage is often not on the same device as the TOE and may be physically hundreds of miles away.  In practice, the TOE encrypts the keys using an approved method before sending them to backup storage.  TOEs may perform cryptographic key escrow in which it entrusts a third party with access to the private or secret keys.  In practice, the TOE protects the keys using a cryptographic key access method agreed upon with the escrow agent before sending them.

The Application Note for this SFR, specifically in the section regarding Key Access Methods, states that if Key Encryption is claimed as a method in FCS_CKM_EXT.3.1, then approved methods such as those included in FCS_COP.1/SKC should be used.  FCS_COP.1.1/SKC currently does not include AES-CFB as an allowed mode.  However, it does include SEED-CFB, HIGHT-CFB and LEA-CFB as allowed modes, so adding AES-CFB may be considered reasonable.  Note that this would add AES-CFB mode for all symmetric key operations, not just key encryption as would be the case for FCS_COP.1.1/KeyEnc in the HCD cPP.

**Comment by tbenkartidtec** [comment]
In a previous note on Apr 19, @Ryuichiro-Ohya-FX provides a PPT file detailing a method of excluding the TPM encryption of keys from the key chain documented in FCS_KYC_EXT.1 and FCS_KYP_EXT.1.  This methodology is consistent with the HCD cPP and SD, since both documents include clarifying text for crypto operations performed by a TPM.  In my opinion, we should consider resolving this issue via that suggested methodology since it does not require any cPP/SD change.

[Crypto] does not include any requirements addressing key chains.  Therefore, while usage of the suggested methodology for excluding TPM-based crypto does not have any direct impact on [Crypto], it is not possible to analyze how usage of that methodology would impact other PPs/cPPs (which would presumably add SFRs to address key chains).
